v4_MIGRATION_GUIDE.md

```diff
@@ -44,6 +48,84 @@ In the `AndroidManifest.xml`, migrate the `intent-filter` from your v3 integrati
+
+In v4, we decoupled the Braintree SDK from Android to offer more integration flexibility. 
+`BraintreeFragment` has been replaced by a `Client` for each respective payment feature. 
+See the below payment method sections for examples of instantiating and using the feature clients. 
+
+## Event Handling
+
+In v3, there were several interfaces that would be called when events occurred: `PaymentMethodNonceCreatedListener`, `ConfigurationListener`, `BraintreeCancelListener`, and `BraintreeErrorListener`.
+In v4, these listeners have been replaced by a callback pattern. 
+
+### Handling `PaymentMethodNonce` Results
+
+For payment methods that do not require leaving the application, the result will be returned via the callback passed into the tokenization method. 
+For example, using the `CardClient`:
+
+```java
+cardClient.tokenize(activity, card, (cardNonce, error) -> {
+  // send cardNonce.getString() to your server or handle error
+});
+```
+
+For payment methods that require a browser switch, the result will be returned as a `BrowserSwitchResult` and should be handled by calling the feature client's `onBrowserSwitchResult()` method in `onResume()`.
+For example, using the `ThreeDSecureClient`:
+
+```java
+@Override
+protected void onResume() {
+  super.onResume();
+
+  BrowserSwitchResult browserSwitchResult = braintreeClient.deliverBrowserSwitchResult(this);
+  if (browserSwitchResult != null) {
+    threeDSecureClient.onBrowserSwitchResult(browserSwitchResult, (threeDSecureResult, error) -> {
+      // send threeDSecureResult.getTokenizedCard().getString() to your server or handle error
+    }); 
+  }
+}
+```
+
+For payment methods that that require an app switch, the result should be handled by calling the feature client's `onActivityResult()` method from your Activity's `onActivityResult()` method.
+For example, using the `VenmoClient`:
+
+```java
+@Override
+protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+  venmoClient.onActivityResult(this, resultCode, data, (venmoAccountNonce, error) -> {
+    // send venmoAccountNonce.getString() to your server or handle error
+  });
+}
+```
+
+Full implementation examples can be found in the payment method feature sections below. 
+
+### Handling Cancellation 
+
+All user cancellations will be returned as a `BraintreeException` with an error message indicating user cancellation to the callback of the invoked method. 
+
+### Handling Errors
+
+Errors will be returned to the callback of the invoked method.
+
+### Fetching Configuration
+
+If you need to fetch configuration, use `BraintreeClient#getConfiguration`.
+
+## Builder Pattern
+
+The builder pattern has been removed in v4 to allow for consistent object creation across Java and Kotlin. 
+Classes have been renamed without the `Builder` postfix, method chaining has been removed, and setters have been renamed with the `set` prefix.
+
+For example, `CardBuilder` in v3 becomes `Card` in v4:
```

Comment:
I see in the CHANGELOG we list all of the builder classes that changed. Would it also be helpful if we list them here too?